docs/content/basics/screenshot_testing.md

```diff
@@ -1,5 +1,86 @@
+     - Drawable caching: the Android system maintains a cache of drawables. If the bitmap of such drawable is shared in several locations, the result could vary depending on whether the bitmap has already been cached in a previous test or not. This introduces flakiness in pixel by pixel comparisons. 
+     - Cached state in Views: some views like ConstraintLayout have their own cache, so that in the next pas they do not have to recalculate anything, what might cause flakiness. 
+     - Dates: if displaying dates that depend on the current time, the screenshots will change on every rerun and fail while comparing to the one taken as reference.
+     - Image Loading from urls: some libraries like Picasso and Glide download images from urls asynchronously under the hood. These makes tests nondeterministic, because we cannot ensure the image loading state of the ImageView before taking the screenshot:
+          1. Random blank image or loading if it was taken before the image was downloaded
+          2. Error image placeholders if the download failed, etc.
+     - RecyclerView prefetching: By default, RecyclerView’s LinearLayoutManager prefetches off screen views outside its viewport while the UI thread is idle between frame. This might cause flakiness if the
+     - FillViewPort measuring: the size of the fillViewPort view might not have been calculated before taking the screenshot. This might result into the fillViewport being cut off, so some children would not be visible
+     - Animation libraries: libraries like lottie, that load an animation into an ImageView from a Json file. Disabling animations do not take effect on this. They happen asynchronously so we cannot guarantee at which stage is the animation before taking the screenshot.
+     - Webview mocking: same problems as asynchronous loading. We cannot ensure the content will be loaded before taking the snapshot 
+
+We'll describe them in detail in their own section in the future.
+
+2. Emulator configuration in all parts involved (*Instrumented Snapshot tests only*)
+     - Emulators freezing on the CI when idle for a long time. The best practice is the same as with Ui tests: close them right after running the tests
+     - Synchronizing emulators start up before running tests. Instrumented Snapshot tests run on every device adb can detect. For that, we need to start all the emulators and wait till they are ready. We can achieve this by using tools like [swarmer] (https://github.com/gojuno/swarmer) or the new official [Gradle Managed Virtual Devices] (https://developer.android.com/studio/preview/features#gmd)
+     - Avoid Insufficient Storage errors by starting adb emulators with the `--wipe_data` option
+
+## Frameworks
+1. Instrumented
+     - [Facebook Screenshot tests](https://github.com/facebook/screenshot-tests-for-android): First screenshot testing framework on Android.
```

Comment:
For the future: would be great to have descriptions here and review of such tools from our side.
As an engineer, I want to use the best and I want to see some advise

Reply:
Definitely. But I would say out of the scope for the MVP. That deserves and article and repo on its own, and I do not know all the frameworks good enough yet